tests/migrations/test_operations.py

```diff
@@ -1692,15 +1692,22 @@ def test_add_field_both_defaults(self):
         field = new_state.models[app_label, "pony"].fields["height"]
         self.assertEqual(field.default, 3)
         self.assertEqual(field.db_default, Value(4))
-        project_state.apps.get_model(app_label, "pony").objects.create(weight=4)
+        pre_pony_pk = (
+            project_state.apps.get_model(app_label, "pony").objects.create(weight=4).pk
+        )
         self.assertColumnNotExists(table_name, "height")
         # Add field.
         with connection.schema_editor() as editor:
             operation.database_forwards(app_label, editor, project_state, new_state)
         self.assertColumnExists(table_name, "height")
+        post_pony_pk = (
+            project_state.apps.get_model(app_label, "pony").objects.create(weight=10).pk
+        )
```

Comment:
This use to fail due with an integrity error due the lack of database level `DEFAULT`

Reply:
Ah I see, now:
 - "pre_pony" is testing the db default is applied to existing columns
 - "post_pony" is testing the db default on a new insert
 - "new_pony" is testing the ORM default

👍

(sorry rubber ducking here the tests are quite verbose 😵)

Reply:
right, the pre/post cases are typical of rolling deployments where you still have _old code_ referring to the model definition that don't have the application level default and must rely on the database level one to work.

In this case it demonstrates that the databases level default is dropped.